Dapper/SqlMapper.cs

```diff
@@ -373,39 +385,46 @@ public static void AddTypeHandlerImpl(Type type, ITypeHandler? handler, bool clo
                 }
             }
 
-            var snapshot = typeHandlers;
-            if (snapshot.TryGetValue(type, out var oldValue) && handler == oldValue) return; // nothing to do
+            // synchronize between callers mutating type-handlers; note that regular query
+            // code may still be accessing the field, so we still use snapshot/mutate/swap;
+            // the synchronize is just to prevent lost writes
+            lock (typeHandlersSyncLock)
+            {
+                var snapshot = typeHandlers;

```

Comment:
This assignment is not necessary.

Reply:
because...?

Reply:
Ah yes, with the new `lock` we know no-one else is swapping this field behind our backs; fine

Reply:
[ea273e0](https://github.com/DapperLib/Dapper/pull/2129/commits/ea273e0297a2af78c3802ed5629651181db7d6d8)